pulp_container/app/viewsets.py

```diff
@@ -595,8 +612,11 @@ def create(self, request, path):
         This methods handles the creation of an upload.
         """
         # TODO add repo push type to distinguish from sync repo type
-        distribution, _ = models.ContainerDistribution.objects.get_or_create(
-            name=path, base_path=path)
+        try:
```

Comment:
this try/expect is surplus here, get_or_create will whether get the exsiting distribution which is what is done on line 619 or create a new one.

Obviously , this will not work in case a base path is already taken and that distribution already serves some other repo. This case we will need to cover separately when we introduce sync/push repos

Reply:
I did not mean to include this. I've removed it in the latest push.